packages/valory/skills/market_creation_manager_abci/rounds.py

```diff
@@ -305,6 +339,114 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         return None
 
 
+class CollectProposedMarketsRound(CollectSameUntilThresholdRound):
+    """CollectProposedMarketsRound"""
+
+    ERROR_PAYLOAD = "ERROR_PAYLOAD"
+    MAX_RETRIES_PAYLOAD = "MAX_RETRIES_PAYLOAD"
+    MAX_APPROVED_MARKETS_REACHED_PAYLOAD = "MAX_APPROVED_MARKETS_REACHED_PAYLOAD"
+    SKIP_MARKET_APPROVAL_PAYLOAD = "SKIP_MARKET_APPROVAL_PAYLOAD"
+
+    payload_class = CollectProposedMarketsPayload
+    synchronized_data_class = SynchronizedData
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            if self.most_voted_payload == self.ERROR_PAYLOAD:
+                proposed_markets_api_retries = cast(
+                    SynchronizedData, self.synchronized_data
+                ).proposed_markets_api_retries
+                synchronized_data = self.synchronized_data.update(
+                    synchronized_data_class=SynchronizedData,
+                    **{
+                        get_name(
+                            SynchronizedData.proposed_markets_api_retries
+                        ): proposed_markets_api_retries
+                        + 1,
+                    },
+                )
+                return synchronized_data, Event.ERROR
+
+            if self.most_voted_payload == self.MAX_RETRIES_PAYLOAD:
+                return self.synchronized_data, Event.MAX_RETRIES_REACHED
+
+            if self.most_voted_payload == self.MAX_APPROVED_MARKETS_REACHED_PAYLOAD:
+                return self.synchronized_data, Event.MAX_APPROVED_MARKETS_REACHED
+
+            if self.most_voted_payload == self.SKIP_MARKET_APPROVAL_PAYLOAD:
+                return self.synchronized_data, Event.SKIP_MARKET_APPROVAL
+
+            synchronized_data = self.synchronized_data.update(
+                synchronized_data_class=SynchronizedData,
+                **{
+                    get_name(
+                        SynchronizedData.collected_proposed_markets_data
+                    ): json.loads(self.most_voted_payload),
+                },
+            )
+            return synchronized_data, Event.DONE
```

Comment:
Why not call `super` instead?

Reply:
Here's an [example](https://github.com/valory-xyz/trader/blob/main/packages/valory/skills/decision_maker_abci/states/tool_selection.py#L33-L47).

And here's an [example](https://github.com/valory-xyz/trader/blob/main/packages/valory/skills/decision_maker_abci/states/decision_receive.py) where the `end_block` is overridden but keeps the patent's logic.

Reply:
fixed.